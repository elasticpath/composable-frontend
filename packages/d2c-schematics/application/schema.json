{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "SchematicsAngularApp",
  "title": "Angular Application Options Schema",
  "type": "object",
  "description": "Generates a new basic application definition in the \"projects\" subfolder of the workspace.",
  "additionalProperties": false,
  "properties": {
    "projectRoot": {
      "description": "The root directory of the new app.",
      "type": "string",
      "visible": false
    },
    "name": {
      "description": "The name of the new app.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the application?"
    },
    "skipTests": {
      "description": "Do not create \"spec.ts\" test files for the application.",
      "type": "boolean",
      "default": false,
      "alias": "S"
    }
  },
  "required": ["name"]
}
