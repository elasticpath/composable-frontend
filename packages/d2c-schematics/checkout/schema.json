{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "SchematicsCheckout",
  "title": "Checkout Options Schema",
  "type": "object",
  "description": "Generates checkout page and supporting functionality in the workspace.",
  "additionalProperties": true,
  "properties": {
    "skipTests": {
      "description": "Do not create \"spec.ts\" test files for the application.",
      "type": "boolean",
      "default": false,
      "alias": "S"
    },
    "skipConfig": {
      "description": "Do not configure Checkout in Elastic Path Commerce Cloud.",
      "type": "boolean",
      "default": false
    },
    "directory": {
      "type": "string",
      "description": "The directory name to create the workspace in."
    },
    "path": {
      "type": "string",
      "format": "path",
      "$default": {
        "$source": "workingDirectory"
      },
      "description": "The path at which to create the component file, relative to the current workspace. Default is a folder with the same name as the component in the project root.",
      "visible": false
    },
    "paymentGatewayType": {
      "description": "The type of payment gateway to use for your checkout experience.",
      "type": "string",
      "enum": ["EP Payments", "None"],
      "x-prompt": "Which payment gateway do you want to use?"
    },
    "epccClientId": {
      "description": "The client id value for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "visible": "false"
    },
    "epccClientSecret": {
      "description": "The client secret value for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "visible": "false"
    },
    "epccEndpointUrl": {
      "description": "The geographical endpoint url for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "visible": "false"
    }
  },
  "required": ["epccClientId", "epccClientSecret", "epccEndpointUrl", "directory"]
}
