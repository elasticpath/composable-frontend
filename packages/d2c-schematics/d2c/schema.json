{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "SchematicsD2C",
  "title": "Elastic Path Commerce Cloud Direct to Consumer (D2C) Options Schema",
  "type": "object",
  "description": "Creates a new Elastic Path Commerce Cloud storefront by combining the workspace and application schematics.",
  "additionalProperties": true,
  "properties": {
    "directory": {
      "type": "string",
      "description": "The directory name to create the workspace in."
    },
    "name": {
      "description": "The name of the new workspace and project.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the new workspace and initial project?"
    },
    "packageManager": {
      "description": "Node package manager to use.",
      "type": "string",
      "enum": ["npm", "yarn", "pnpm"],
      "default": "npm"
    },
    "epccClientId": {
      "description": "The client id value for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "x-prompt": "What is the client id for your store?"
    },
    "epccClientSecret": {
      "description": "The client secret value for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "x-prompt": "What is the client secret for your store?"
    },
    "epccEndpointUrl": {
      "description": "The geographical endpoint url for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "default": "api.moltin.com",
      "x-prompt": "What is the endpoint for your store?"
    },
    "skipInstall": {
      "description": "Do not install dependency packages.",
      "type": "boolean",
      "default": false
    },
    "skipGit": {
      "description": "Do not initialize a git repository.",
      "type": "boolean",
      "default": false,
      "alias": "g"
    },
    "commit": {
      "description": "Initial git repository commit information.",
      "oneOf": [
        { "type": "boolean" },
        {
          "type": "object",
          "properties": {
            "name": {
              "type": "string"
            },
            "email": {
              "type": "string",
              "format": "email"
            },
            "message": {
              "type": "string"
            }
          },
          "required": ["name", "email"]
        }
      ],
      "default": true
    },
    "skipTests": {
      "description": "Do not generate \"spec.ts\" test files for the new project.",
      "type": "boolean",
      "default": false,
      "alias": "S"
    },
    "plpType": {
      "description": "The type of product list page to use.",
      "type": "string",
      "enum": ["Algolia", "None"]
    }
  },
  "required": ["name", "epccClientId", "epccClientSecret", "epccEndpointUrl"]
}
