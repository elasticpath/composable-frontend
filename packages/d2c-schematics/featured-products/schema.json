{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "FeaturedProductsSchema",
  "title": "Featured Products Schematics Schema",
  "type": "object",
  "additionalProperties": false,
  "properties": {
    "path": {
      "type": "string",
      "format": "path",
      "$default": {
        "$source": "workingDirectory"
      },
      "description": "The path at which to create the component file, relative to the current workspace. Default is a folder with the same name as the component in the project root.",
      "visible": false
    },
    "featured-node-id": {
      "description": "The id of the EPCC node entity you want to use to power the featured product component",
      "type": "string",
      "x-prompt": "What is the id of the node you want to use for featured products?"
    }
  },
  "required": ["featured-node-id"]
}
