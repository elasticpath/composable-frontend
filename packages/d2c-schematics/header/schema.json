{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "HeaderSchema",
  "title": "Header Schematics Schema",
  "type": "object",
  "additionalProperties": false,
  "properties": {
    "path": {
      "type": "string",
      "format": "path",
      "$default": {
        "$source": "workingDirectory"
      },
      "description": "The path at which to create the component file, relative to the current workspace. Default is a folder with the same name as the component in the project root.",
      "visible": false
    },
    "search": {
      "description": "Should include the search functionality",
      "type": "boolean",
      "default": true
    }
  }
}
