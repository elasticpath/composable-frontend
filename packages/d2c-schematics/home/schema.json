{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "HomeSchema",
  "title": "Home Schematics Schema",
  "type": "object",
  "additionalProperties": false,
  "properties": {
    "path": {
      "type": "string",
      "format": "path",
      "$default": {
        "$source": "workingDirectory"
      },
      "description": "The path at which to create the component file, relative to the current workspace. Default is a folder with the same name as the component in the project root.",
      "visible": false
    },
    "components": {
      "description": "The home page components to include.",
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": ["PromotionBanner", "FeaturedProducts"],
        "type": "string"
      },
      "default": [],
      "x-prompt": "What home page components would you like to include?"
    }
  }
}
