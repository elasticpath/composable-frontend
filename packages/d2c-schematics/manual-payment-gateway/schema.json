{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "ManualPaymentGatewayCheckout",
  "title": "Manual Payment Gateway Options Schema",
  "type": "object",
  "description": "Generates checkout page and supporting functionality in the workspace.",
  "additionalProperties": true,
  "properties": {
    "skipTests": {
      "description": "Do not create \"spec.ts\" test files for the application.",
      "type": "boolean",
      "default": false,
      "alias": "S"
    },
    "skipConfig": {
      "description": "Do not configure Checkout in Elastic Path Commerce Cloud.",
      "type": "boolean",
      "default": false
    },
    "directory": {
      "type": "string",
      "description": "The directory name to create the workspace in."
    },
    "path": {
      "type": "string",
      "format": "path",
      "$default": {
        "$source": "workingDirectory"
      },
      "description": "The path at which to create the component file, relative to the current workspace. Default is a folder with the same name as the component in the project root.",
      "visible": false
    },
    "epccClientId": {
      "description": "The client id value for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "visible": "false"
    },
    "epccClientSecret": {
      "description": "The client secret value for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "visible": "false"
    },
    "epccEndpointUrl": {
      "description": "The geographical endpoint url for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "visible": "false"
    }
  },
  "required": ["epccClientId", "epccClientSecret", "epccEndpointUrl", "directory"]
}
