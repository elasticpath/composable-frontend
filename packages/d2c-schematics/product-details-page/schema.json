{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "SchematicsProductDetails",
  "title": "Product Details Options Schema",
  "type": "object",
  "description": "Generates a product details page in the workspace.",
  "additionalProperties": false,
  "properties": {
    "skipTests": {
      "description": "Do not create \"spec.ts\" test files for the application.",
      "type": "boolean",
      "default": false,
      "alias": "S"
    },
    "path": {
      "type": "string",
      "format": "path",
      "$default": {
        "$source": "workingDirectory"
      },
      "description": "The path at which to create the component file, relative to the current workspace. Default is a folder with the same name as the component in the project root.",
      "visible": false
    }
  }
}
