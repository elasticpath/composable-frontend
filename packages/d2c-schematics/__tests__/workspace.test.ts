import { SchematicTestRunner } from "@angular-devkit/schematics/testing"
import path from "path"
import { Schema as WorkspaceOptions } from "../workspace/schema"
import { parse as parseJson } from "jsonc-parser"

const collectionPath = path.join(__dirname, "../collection.json")

describe("Workspace Schematic", () => {
  const schematicRunner = new SchematicTestRunner("schematics", collectionPath)
  const defaultOptions: WorkspaceOptions = {
    name: "foo",
    epccClientId: "123",
    epccClientSecret: "456",
    epccEndpointUrl: "euwest.api.elasticpath.com",
  }

  it("workspace schematic should create files for workspace", async () => {
    const options = { ...defaultOptions }

    try {
      const tree: any = await schematicRunner
        .runSchematicAsync("workspace", options)
        .toPromise()
      const files = tree.files
      expect(files.sort()).toMatchSnapshot()
    } catch (err) {
      throw new Error(err)
    }
  })

  it("workspace schematic should not configure testing when tests=false", async () => {
    const tree = await schematicRunner.runSchematic("workspace", {
      ...defaultOptions,
      tests: false,
    })

    const { scripts, devDependencies } = parseJson(
      tree.readContent("package.json").toString(),
    )
    expect(scripts).not.toContainKeys(["test", "test:watch"])
    expect(devDependencies).not.toContainKeys([
      "jest",
      "@testing-library/jest-dom",
      "@testing-library/react",
    ])
    const files = tree.files
    expect(files).not.toIncludeAnyMembers([
      "playwright.config.ts.template",
      ".env.test.template",
      "jest.config.ts.template",
    ])
  })
})
