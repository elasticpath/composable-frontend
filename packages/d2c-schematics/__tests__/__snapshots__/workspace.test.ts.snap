// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workspace Schematic workspace schematic should create files for workspace 1`] = `
[
  "/.composablerc",
  "/.env.example",
  "/.env.local",
  "/.env.test",
  "/.eslintrc.json",
  "/.gitignore",
  "/.lintstagedrc.js",
  "/.prettierignore",
  "/.prettierrc",
  "/README.md",
  "/license.md",
  "/next-env.d.ts",
  "/next.config.js",
  "/package.json",
  "/playwright.config.ts",
  "/postcss.config.js",
  "/tailwind.config.ts",
  "/tsconfig.json",
  "/vite.config.ts",
]
`;
